website/content/docs/job-specification/template.mdx

```diff
@@ -365,6 +365,179 @@ EOH
 }
 ```
 
+### Nomad Variables
+
+Nomad [variables] can be queried using the `nomadVar`, `nomadVarSafe`,
+`nomadVarList`, `nomadVarListSafe`, and `nomadVarExists` functions.
+
+#### `nomadVarList` and `nomadVarListSafe`
+
+These functions can be used to list the paths of Nomad variables available to
+the task based on its [workload identity]. You can provide an optional prefix to
+filter the path list by as a parameter.
+
+The list functions return a slice of `NomadVarMeta` type:
+
+```golang
+type NomadVarMeta struct {
+	Namespace, Path          string
+	CreateIndex, ModifyIndex uint64
+	CreateTime, ModifyTime   nanoTime
+}
+```
+
+The `nanoTime` type contains Unix nanoseconds since the epoch. Its string method
+prints it out as a formatted date/time value. It also has a `Time` method that
+can be used to retrieve a Go [`time.Time`] for further manipulation.
+
+`NomadVarMeta` objects print their `Path` value when used as a string. For
+example, these two template blocks produce identical output:
+
+```hcl
+template {
+  data        = <<EOH
+{{ range nomadVarList }}
+  {{ . }}
+{{ end }}
+}
+EOH
+}
+
+template {
+  data        = <<EOH
+{{ range nomadVarList }}
+  {{ .Path }}
+{{ end }}
+}
+EOH
+}
+```
+
+You can provide a prefix to filter the path list as an optional parameter to the
+`nomadVarList` function.
+
+```hcl
+template {
+  data        = <<EOH
+{{ range nomadVarList "path/to/filter" }}
+  {{ . }}
+{{ end }}
+}
+EOH
+}
+```
+
+By default the `nomadVarList` will list variables in the same namespace as the
+task. The path filter can change the a namespace by adding a suffix separated
+by the `@` character:
+
+```hcl
+template {
+  data        = <<EOH
+{{ range nomadVarList "path/to/filter@example_namespace" }}
+  {{ . }}
+{{ end }}
+}
+EOH
+}
+```
+
+The `nomadVarListSafe` function works identically to `nomadVarList`, but refuses
+to render the template if the variable list query returns blank/empty data.
+
+
+#### `nomadVar` and `nomadVarSafe`
```

Comment:
No nomadVarSafe
```suggestion
#### `nomadVar`
```

Reply:
Oh! Removed from the first paragraph as well.